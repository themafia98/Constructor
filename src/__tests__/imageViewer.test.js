import React from 'react';
import renderer from 'react-test-renderer';
import ImageViewer from '../components/imageViewer/imageViewer';

const path = "https://images.unsplash.com/";
let imageurl = "photo-1504215680853-026ed2a45def?ixlib=rb-1.2.1&amp;q=80&amp;fm=jpg&amp;";
imageurl = imageurl + "crop=entropy&amp;cs=tinysrgb&amp;w=1080&amp;fit=max&amp;ixid=eyJhcHBfaWQiOjgwNjg0fQ";

const imagePath = path + imageurl;

let ImageViewerComponent = renderer.create(
    <ImageViewer path ={imagePath} />
)

test('imageViewer.js test ', function(){

    let snapshotComponent = ImageViewerComponent.toJSON();
    expect(snapshotComponent).toMatchSnapshot();

    const setCurrentImage = ImageViewerComponent.root.find(item => item.props.className === 'ImageViewer');
    setCurrentImage.props.onClick({stopPropagation: () => null});

    snapshotComponent = ImageViewerComponent.toJSON();
    expect(snapshotComponent).toMatchSnapshot();
});