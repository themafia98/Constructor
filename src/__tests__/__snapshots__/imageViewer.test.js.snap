// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`imageViewer.js test  1`] = `
<div
  className="ImageViewer"
  onClick={[Function]}
>
  <img
    alt="fullImage"
    src="https://images.unsplash.com/photo-1504215680853-026ed2a45def?ixlib=rb-1.2.1&amp;q=80&amp;fm=jpg&amp;crop=entropy&amp;cs=tinysrgb&amp;w=1080&amp;fit=max&amp;ixid=eyJhcHBfaWQiOjgwNjg0fQ"
  />
</div>
`;

exports[`imageViewer.js test  2`] = `
<div
  className="ImageViewer"
  onClick={[Function]}
>
  <img
    alt="fullImage"
    src="https://images.unsplash.com/photo-1504215680853-026ed2a45def?ixlib=rb-1.2.1&amp;q=80&amp;fm=jpg&amp;crop=entropy&amp;cs=tinysrgb&amp;w=1080&amp;fit=max&amp;ixid=eyJhcHBfaWQiOjgwNjg0fQ"
  />
</div>
`;
